### ISSUE TITLE
BOTH THEMES: Forge deploy appears to not be working?

### ISSUE THREAD

**Issue by KathJohnson** [meetinghouse/cms#224]
I am trying to deploy code to the site testdeploy.corbettresearchgroup.com, and Forge is saying it isn't deploying successfully, so now I'm not sure of the state of the code. I deployed code to lighthemetest as well, and I got an error messageIt actually appears to me that the code HAS deployed (because the favicon did change, and that's the kind of change I'm hoping to see). So this is more of a question than a troubleshoot right now -- Andy, should I worry about this error message from Forge?

Error message resulting from push to testdeploy
![screenshot 2016-02-24 at 11 31 35 am](https://cloud.githubusercontent.com/assets/5922040/13292741/f30e646a-daea-11e5-9793-4e42d1295a58.png)

Error message resulting from push to lighthemetest
![screenshot 2016-02-24 at 11 37 20 am](https://cloud.githubusercontent.com/assets/5922040/13292760/02f7441e-daeb-11e5-9f7e-810d2ea562bb.png)

**Comment by KathJohnson** [comment]
Andy, do you have any ideas on why I might be having errors on deploying code through Forge? I have a new error message today when attempting to deploy code to lightthemetest.corbettresearchgroup.com

Is this something you're able to troubleshoot? I'm hoping you have the answer for this so we can implement all the good changes and updates that you and John B. have been working on.

![screenshot 2016-03-08 at 1 51 20 pm](https://cloud.githubusercontent.com/assets/5922040/13612384/37b6ab00-e535-11e5-9469-fd14f17a7497.png)

**Comment by acavanagh** [comment]
Most of it is because there's a typo here:
https://github.com/meetinghouse/cms/blob/blog/app/database/migrations/2014_03_12_032407_create_pages_table.php#L20
that
was introduced in this commit:
https://github.com/meetinghouse/cms/commit/a61e05dfb5c8fadaf1200e89c2bc8f270d6689c0

**Comment by acavanagh** [comment]
there's another issue on at least one of the sites where you can't update
it because John and/or Barry were working directly on the server, and the
deploy script won't overwrite their changes.

**Comment by jboz62** [comment]
Yes, I was speculating about that at our meeting today. Is there a way to reconcile that or force it to overwrite? Both Barry and I have been working via SFTP because we haven’t been able to set up locally. I didn’t have any luck getting set up locally so the only way I can test is to use SFTP. I’m pretty buried right now, so it’s hard to find time to figure out how to set that up. If you have some time and are willing, I would appreciate some help getting that going so I can work the right way on these. 

Thanks for the input, much appreciated.

**Comment by acavanagh** [comment]
I've got a little time on sunday morning if you want to go over setting up locally.  If you're on a mac you can use either vagrant/homestead or mamp.  If you're on windows you pretty much have to set up a virtual machine if you want to run the code locally.  You can emulate what you're doing now without the git and deployment headaches by just cloning down the repository locally, pushing your work to git, and then deploying to the test site via forge.

Working directly on the server is going to block deployments, foul up version control, and end up with changes being overwritten.

**Comment by jboz62** [comment]
Thanks, sunday will work. I’m on a mac using mamp. 
> You can emulate what you're doing now without the git and deployment headaches by just cloning down the repository locally, pushing your work to git, and then deploying to the test site via forge.
> 
I have a local copy I’ve been using for git sitting in MAMP, just need to get it working for preview. I’ll try the method you outlined here using forge. Is there a way to make forge overwrite the changes that were made or do we have to nuke those sites and set up clean installs? The changes are all in github now, so it’s ok if they are overwritten on those sites.

**Comment by meetinghouse** [comment]
We could delete the site testdeploy as far as I'm concerned, and then I can re-deploy it anew for future use. As far as I know, any changes that were made directly to it have been merged into the Git code. Am I correct about that? 

Establishing a process is still great, but at least this one troublesome site doesn't need to continue being troublesome from where I'm sitting.

**Comment by meetinghouse** [comment]
I merged the pull request to fix the typo and tried to deploy again just now to both lightthemetest and testdeploy. I received errors for both of them. That may not be helpful information, but I wanted to document it.

**Comment by jboz62** [comment]
Yes, I think that’s a good plan, was thinking the same.

**Comment by jboz62** [comment]
Looks to me like all the deploys are working except for “testdeploy” and “lightthemetest”, but we don’t need to use those, right? We can just delete and recreate new test sites. If the other sites deploy correctly, which is looks like they do, we’re in good shape. Thoughts?

**Comment by jboz62** [comment]
Katherine, I went through and cleaned up testdeploy and lightthemetest, then I did a deploy on lightthemetest which went through fine. You should be good to deploy to testdeploy also. Once the latest pull request is merged and deployed, we should be ready to start rolling out all the latest changes to the sites.

**Comment by KathJohnson** [comment]
I was able to deploy to testdeploy, no more errors! Thank you, all.
